detectable/src/main/java/com/blackduck/integration/detectable/detectables/go/gomod/process/GoModDependencyManager.java

```diff
@@ -40,19 +41,30 @@ private Map<String, Dependency> convertModulesToDependencies(List<GoListAllData>
             String name = Optional.ofNullable(module.getReplace())
                 .map(ReplaceData::getPath)
                 .orElse(module.getPath());
-            String version = Optional.ofNullable(module.getReplace())
+            String kbCompatibleVersion = Optional.ofNullable(module.getReplace())
                 .map(ReplaceData::getVersion)
                 .orElse(module.getVersion());
-            if (version != null) {
-                version = handleGitHash(version);
-                version = removeIncompatibleSuffix(version);
+            if (kbCompatibleVersion != null) {
```

Comment:
Perhaps a small method for the contents of this if along with some comments on it that describe what is happening might help to maintain this.